devtools/gradle/gradle-extension-plugin/src/main/java/io/quarkus/extension/gradle/utils/QuarkusGradleUtils.java

```diff
@@ -0,0 +1,34 @@
+package io.quarkus.extension.gradle.utils;
+
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+
+import io.quarkus.bootstrap.model.ApplicationModel;
+import io.quarkus.bootstrap.model.gradle.ModelParameter;
+import io.quarkus.bootstrap.model.gradle.impl.ModelParameterImpl;
+import io.quarkus.gradle.tooling.GradleApplicationModelBuilder;
+import io.quarkus.runtime.LaunchMode;
+
+public class QuarkusGradleUtils {
+
+    public static Path serializeAppModel(ApplicationModel appModel, Task context, boolean test) throws IOException {
+        final Path serializedModel = context.getTemporaryDir().toPath()
+                .resolve("quarkus-app" + (test ? "-test" : "") + "-model.dat");
+        try (ObjectOutputStream out = new ObjectOutputStream(Files.newOutputStream(serializedModel))) {
+            out.writeObject(appModel);
+        }
+        return serializedModel;
+    }
```

Comment:
It looks like we have this kind of utility API in the application plugin, right? Should we move it to the model module?

Reply:
Definitely, I'm moving that